bdk_core/src/coin_select.rs

```diff
@@ -1,77 +1,128 @@
 use crate::{collections::BTreeSet, Vec};
 use bitcoin::{LockTime, Transaction, TxOut};
 
-const TXIN_BASE_WEIGHT: u32 = (32 + 4 + 4) * 4;
+pub const TXIN_FIXED_WEIGHT: u32 = (32 + 4 + 4) * 4;
 
 #[derive(Debug, Clone)]
-pub struct CoinSelector {
-    candidates: Vec<WeightedValue>,
+pub struct CoinSelector<'a> {
+    candidates: Vec<InputCandidate>,
     selected: BTreeSet<usize>,
-    opts: CoinSelectorOpt,
+    opts: &'a CoinSelectorOpt,
```

Comment:
I think you can just Copy this in each time. I don't think it needs to be a reference.